python/arcticdb/version_store/_store.py

```diff
@@ -1498,7 +1520,8 @@ def _get_read_query(self, date_range: Optional[DateRangeInput], row_range, colum
             read_query.row_filter = _normalize_dt_range(date_range)
 
         if row_range is not None:
-            read_query.row_range = row_range
+            total_n_rows = self.get_num_rows(symbol=symbol, as_of=as_of, **kwargs)
```

Comment:
Note to reviewers: [`get_description`](https://docs.arcticdb.io/api/library#arcticdb.version_store.library.Library.get_description) is an alternative, but this is unreachable in this scope.

Reply:
The reason I did this change was to get ride of `calculate_row_filter` and `SignedRowRange` in `cpp/arcticdb/pipeline/query.hpp`.

I first tried to have this logic in the C++ side but I do not know what is the best way to get the total number of lines as soon as possible unfortunately.

Reply:
> it will be read twice with this implementation

How costly is this? I do not think we should optimize something that is not measured.

Reply:
Reverted back with e2b97e9.